### PR TITLE
created tests for add-on-store component

### LINKED ISSUE #1038: Code Coverage: Create tests for src/components/AddOn/core/AddOnStore/AddOnStore.tsx
We will need unittests done for all methods, classes and functions found in this file.

Any widgets, components, modals referenced in this file must also have unittests done

### PR Acceptance Criteria

- When complete this file must show at least **100%** coverage when merged into the code base. 
- [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin?search=&displayType=list)
- The PR will show a report for the code coverage for the file you have added. You can use that as a guide
- Please coordinate **issue assignment** and **PR reviews** with the contributors listed in this issue https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by ALOK9442**
Fixes #1038 

Did you add tests for your changes?
Yes

Snapshots/Videos:
Not required

If relevant, did you update the documentation?

No documentation update required

Summary

I added unit tests for src/components/AddOn/core/AddOnStore/AddOnStore.tsx

Does this PR introduce a breaking change?
No

Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?

Yes, I have read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by ALOK9442** [comment]
Hi @palisadoes  in context of my previous pr #1363 I have raised pr from updated upstream/develop

**Comment by palisadoes** [comment]
@SiddheshKukade This is for additional plugin testing

**Comment by SiddheshKukade** [comment]
Please fix lint problems and other errors

**Comment by ALOK9442** [comment]
hello @SiddheshKukade  I have fixed the lint problems and other errors, You can check it for further changes if required (any)

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx [hunk 1]
```diff
@@ -30,31 +30,338 @@ const httpLink = new HttpLink({
...
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      fireEvent.click(screen.getByText('Install'));
+    });
+    waitFor(() => {
+      expect(screen.getByText('Install')).toBeInTheDocument();
```
Comment (beingnoble03) [inline src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:364]: How does this imply "A plugin is installed?"

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx [hunk 2]
```diff
@@ -30,31 +30,338 @@ const httpLink = new HttpLink({
...
-    expect(getByTestId('AddOnEntryStore')).toBeInTheDocument();
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
+      fireEvent.click(screen.getByText('Example Plugin'));
+    });
+
+    waitFor(() => {
+      expect(screen.getByText('Example Plugin')).toBeInTheDocument();
```
Comment (beingnoble03) [inline src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:290]: How does this imply "correct Redux action is dispatched"?

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx [hunk 3]
```diff
@@ -30,31 +30,338 @@ const httpLink = new HttpLink({
...
+              <AddOnStore />
+            </I18nextProvider>
+          </BrowserRouter>
+        </Provider>
+      </ApolloProvider>
+    );
+
+    await waitFor(() => screen.getByTestId('AddOnEntryStore'));
+
+    waitFor(() => {
```
Comment (beingnoble03) [inline src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:265]: I don't think we need to use `waitFor` here?

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
@ALOK9442 Please address the comments.

Also, I don't think we need to enclose `expect` func in `waitFor` func in test cases. Please let me know.

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx [hunk 4]
```diff
@@ -30,31 +30,338 @@ const httpLink = new HttpLink({
   },
 });
 
-const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
+const client = new ApolloClient({
```
[the remark this answers is not in the document]
Comment (ALOK9442) [reply src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:33]: fixing it as I forgot to correct it after changes

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx (hunk 2 above)
Comment (ALOK9442) [reply src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:290]: it is working as it is expected to be

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx (hunk 1 above)
Comment (ALOK9442) [reply src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:364]: It is working as it is expected, pls let me know if I am not getting your doubt.

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx (hunk 3 above)
Comment (ALOK9442) [reply src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:265]: I am using waitFor here to make sure the data gets fetched as per the code,

**Comment by ALOK9442** [comment]
@beingnoble03 thanks for adding comment, I enclosed expect in waitFor to make sure the DOM gets updated after any events.

also It is giving me error if I am not waiting for it 
this error is of a specific test - "TestingLibraryElementError: Unable to find an element with the text: Error loading data. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible."

Please let me know.

**Comment by beingnoble03** [comment]
@ALOK9442 oh I see. You're using `screen.Get....` inside the expect. That's why you need `waitFor` outside the `expect` func. Fine. This works.

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx (hunk 2 above)
Comment (beingnoble03) [reply src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:290]: See the test says `clicks on an AddOnEntry and dispatches the correct Redux action`, but you are testing if "Example Plugin" is present on the screen. How are these even related?

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx (hunk 1 above)
Comment (beingnoble03) [reply src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:364]: Same explanation as above.

**Comment by ALOK9442** [comment]
@beingnoble03 yes, also can you please help me with this error of graphql happening at run check

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx (hunk 1 above)
Comment (ALOK9442) [reply src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:364]: yes

File: src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx (hunk 2 above)
Comment (ALOK9442) [reply src/components/AddOn/core/AddOnStore/AddOnStore.test.tsx:290]: oh I see let me see it in my code.

**Comment by beingnoble03** [comment]
@ALOK9442 You can ignore that check for now. There's some issue in our github actions.

**Comment by noman2002** [comment]
@ALOK9442 The coverage only gets to 30% after this. Please get it to 100%.

**Comment by palisadoes** [comment]
Please fix the failing tests
- introspection

**Comment by ALOK9442** [comment]
working on it

**Comment by SiddheshKukade** [comment]
Please fix introspection. So that we can proceed.

**Comment by noman2002** [comment]
Closing due to inactivity.

**Pull request closed without merging**